config/crd/patches/validation_in_linodemachines.yaml

```diff
@@ -0,0 +1,17 @@
+# The following patch adds additional constraints after the built-in name validation for the CRD
+- op: add
+  path: /spec/versions/0/schema/openAPIV3Schema/properties/metadata/properties
+  value:
+    name:
+      type: string
+      x-kubernetes-validations:
+      - rule: 3 <= size(self) && size(self) <= 64
+        message: linode instance labels must be between 3..64 characters
+      # TODO: Combine these into one regex to validate ALL constraints on Linode instance labels at once?
```

Comment:
would these commented-out rules be covered by the k8s default `metadata.name` validation already?

Reply:
Not necessarily. For example, [RFC 1123](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names) allows double-hypens (which is disallowed by some Linode resource labels).

Reply:
yeah, I'm talking specifically lines 11-15 here

Reply:
Yes, they should be. However, I opted to add in a 1:1 mapping with all the Linode label validation rules just to make this complete in itself and easier to read.

Reply:
that is fair, if we are going to leave it commented out could we just write something describe why we are keeping them in there but not enabling them?

Reply:
I've cleaned up the commented-out rules and updated the error messages. I've also updated the "Testing" section of the Description to better show what the error messages from the API server would look like.

Technically, this rule in the `LinodeMachine` and `LinodeMachineTemplate` definition is covered by the default validation, but I've left it in for clarity and context: 
```yaml
      - rule: self.matches('^[[:alnum:]]([-_.[:alnum:]]+[[:alnum:]])*$')
        message: >-
          custom validation:
          linode instance: labels:
          must begin and end with an alphanumeric character,
          may only consist of alphanumeric characters, hyphens (-), underscores (_) or periods (.),
          cannot have two hyphens (--), underscores (__) or periods (..) in a row,
          regex used for validation is: '^[[:alnum:]]([-_.[:alnum:]]+[[:alnum:]])*$',
          see: https://www.linode.com/docs/api/linode-instances/#linode-create
```

(and in case we would like to optimize this in the future and combine all the rules into one regex idk 🤷‍♀️)